packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1411,22 +1437,28 @@ abstract class BuildableElement extends Element {
     assert(!_dirty);
   }
 
-  /// Called by [rebuild] after [rebuild] has checked that this element indeed
-  /// needs to build and is still active.
-  ///
-  /// This function is called if this element is marked as needing to be built
-  /// (see [markNeedsBuild]). For example, if [State.setState] is called on its
-  /// associated [State] object or if this element depends on an
-  /// [InheritedElement] that has itself changed.
+  /// Called by rebuild() after the appropriate checks have been made.
   void performRebuild();
 
   @override
-  void dependenciesChanged(Type affectedWidgetType) {
+  void dependenciesChanged() {
+    assert(_active);
     markNeedsBuild();
   }
 
   @override
+  void activate() {
+    bool rebuild = ((_dependencies != null && _dependencies.length > 0) || _hadUnsatisfiedDependencies);
```

Comment:
Done. Also renamed the variable to shouldRebuild to avoid stomping on the rebuild method.